### PR TITLE
feat: add connection rate limit setting

### PR DESCRIPTION
**Pull request by e2corporation**
### :package: Config-UI / Data Integrations / Connections - **Rate Limit** 

- [x] Add **Rate Limit** Connection Form Field
- [x] Update Connection Manager Hook (Define `rateLimit`)
- [x] Update Component Properties
- [x] Test Rate Limit Setting

### Description

This PR adds the **Rate Limit** (per hour) numeric setting for the **Connections** definition for All Data Providers (Plugins). The default value is `0`. NOTE: It's property is `NOT` included when _Testing_ a connection.

### Does this close any open issues?
#931

### Screenshots
Screen Shot 2022-08-19 at 7 19 03 PM

Screen Shot 2022-08-19 at 7 25 21 PM

### CODE REVIEW

**Comment by e2corporation** [comment]
@klesh @warren830 Is BE currently storing rateLimit? It's still returning `0` even though request payload sends a different value.

**Comment by e2corporation** [comment]
:warning: NOTE: This PR may conflict with PR #2782. The changes on PR's #2782 `blueprint-settings.jsx` should be preferred.

File: config-ui/src/data/NullConnection.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ const NullConnection = {
   name: null,
   endpoint: null,
   proxy: null,
+  rateLimit: 0,
```
Comment (klesh) [inline config-ui/src/data/NullConnection.js:24]: it should be ratelimitPerHour

Comment (e2corporation) [reply config-ui/src/data/NullConnection.js:24]: I looked at the code rateLimit is the key being used. The API response even sends back "rateLimit"

Comment (e2corporation) [reply config-ui/src/data/NullConnection.js:24]: or only on the request payload it needs to be `ratelimitPerHour`?

Comment (e2corporation) [reply config-ui/src/data/NullConnection.js:24]: @klesh changed request payload key to `rateLimitPerHour`

Comment (e2corporation) [reply config-ui/src/data/NullConnection.js:24]: @klesh Internally on UI we use `rateLimit`. It is defined correctly in `NullConnection`.

**Comment by e2corporation** [comment]
@klesh PR Updated ready to go.

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**